shared/charts/ssl-cert-check/templates/cronjob.yaml

```diff
@@ -0,0 +1,113 @@
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: {{ template "ssl-cert-check.name" . }}
+  labels:
+    app: {{ template "ssl-cert-check.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  schedule: "{{ .Values.cronJobSchedule }}"
+  successfulJobsHistoryLimit: 1
+  failedJobsHistoryLimit: 1
+  concurrencyPolicy: Replace
+  startingDeadlineSeconds: 200
+  jobTemplate:
+    spec:
+      activeDeadlineSeconds: {{ .Values.activeDeadlineSeconds }}
+      template:
+        metadata:
+          labels:
+            app: {{ template "ssl-cert-check.name" . }}
+            release: {{ .Release.Name }}
+          {{- if .Values.istio.enabled }}
+          annotations:
+            # This annotation is needed to allow traffic to metadata.google.internal
+            # until Istio's issue with FQDNs in ServiceEntries is not resolved:
+            # https://github.com/istio/istio/issues/14404
+            traffic.sidecar.istio.io/excludeOutboundIPRanges: "169.254.169.254/32"
+          {{- end }}
+        spec:
+          shareProcessNamespace: true
+          containers:
+          - name: ssl-cert-check
+            image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+            imagePullPolicy: {{ .Values.image.pullPolicy }}
+            command:
+              - /bin/sh
+              - -c
+              - timeout {{ sub .Values.activeDeadlineSeconds 15 }} /entrypoint.sh || true
```

Comment:
Why is the `timeout` required here?
Won't the return value be masked by `|| true` construct, therefore making it impossible for Kubernetes & operators to know whether the job succeeded or failed?

Reply:
This is because running a task that spawns a daemon does not really fit into `CronJob` mechanics. Anyway, this is not actual anymore, since I adopted Locust Stackdriver client for `ssl-cert-check`.